router/handlers.go

```diff
@@ -2,15 +2,59 @@ package router
 
 import (
 	"fmt"
+	"html/template"
 	"log"
 	"net/http"
 )
 
+type PageSettings struct {
+	PageTitle string
+	filename  string
+	filepath  string
+}
+
+func NewPageSettings(pageTitle string, filename string) *PageSettings {
```

Comment:
I spy with my little eye a C-developer! This and `CreateTemplate` are _great_ wrappers, I applaud your instinct, but Go has already solved this problem in `html/template`.

Use this as a reference:
https://gowebexamples.com/templates/

You are currently: creating a new struct, creating a new template, executing a template _every request_. You're also reinventing a wheel that's already provided for you.